### ISSUE TITLE
Remove meaningless lines from doc

### ISSUE THREAD

**Issue by kohr-h** [odlgroup/odl#68]
It's probably mostly me who is responsible for the existence of lines like `Initialize a new instance.`, due to an attempt to slavishly follow [PEP 257](https://www.python.org/dev/peps/pep-0257/) and, to be more precise, the recommendations of the [pep257](https://pypi.python.org/pypi/pep257) tool. One particular warning is raised if, among others, 

- there is no summary line (a single line or a line separated by a blank line from the rest)
- the summary line does not end with a dot
- the first word ends with an 's' indicating non-imperative mood (e.g. "Returns ..." in contrast to the preferred "Return ...")

In the case of `__init__()`, though, I think we should consider deleting the summary line since it adds exactly zero value to the documentation and in some cases interferes with e.g. Spyder's object inspector output in an ugly manner, appearing between the class docstring and the `__init__()` docstring.

Further examples of meaningless documentation are (to be completed):

- function or method without parameters, clear from the signature already:
  ```python
  def myfunc():
      """Do something.
  
      Parameters
      ----------
      None
  
      Returns
      -------
      something : `object`
  ```
  Here we should simply delete the `Parameters` section.

**Comment by adler-j** [comment]
* Also all the cases of explaining python to the user in magic methods:

    ```
    def __add__(self, other):
        """``op.__add__(other) <==> op + other``."""
        pass
    ```

    Worst case this could be simplified to `op + other`.

**Comment by kohr-h** [comment]
> Regarding the ``__init__`` method it could be usefull in some cases.

The docstring itself, yes. The information that this function initializes a new instance, no.



> All the cases of explaining python to the user in magic methods: [...]

That's actually copied behavior from NumPy. We can consider removal in cases where there is no extra information, sure. These methods could IMO go without docstring since they're not public anyway, at least not directly. Instead we could add a small section in the package doc with a link to [this great summary](http://rafekettler.com/magicmethods.html).

In general I suggest that we leave the summary line empty and start directly with `Parameters` for those "obvious" functions.

**Comment by kohr-h** [comment]
Compare 
![with `__init__` summary](https://cloud.githubusercontent.com/assets/5030250/11320646/5c4b4974-90a0-11e5-94c8-84201dcb2c5f.png) and 
![without](https://cloud.githubusercontent.com/assets/5030250/11320647/704875c8-90a0-11e5-993b-ab280b043448.png). I find the extra line in the first one very annoying.

**Comment by adler-j** [comment]
Does this have a good definition of done or is it ongoing?

**Comment by kohr-h** [comment]
No, actually this has been deprecated by the closed issue #77, where it became apparent that an empty summary screws up the LaTeX doc.

**Comment by kohr-h** [comment]
Apart from the summary, I think the "empty" parameter sections can be fixed in a major doc overhaul which is anyway necessary, but continuous work.
